truss/contexts/image_builder/serving_image_builder.py

```diff
@@ -488,7 +488,9 @@ def _render_dockerfile(
             live_reload=config.live_reload,
             data_dir_exists=data_dir.exists(),
             bundled_packages_dir_exists=bundled_packages_dir.exists(),
-            truss_hash=directory_content_hash(self._truss_dir),
```

Comment:
this logic is duplicated between here and truss_handle. Both have access to a TrussSpec, could we consider moving the logic for generating the truss_hash in the TrussSpec? (so it would be `self._spec.truss_hash()`, which could then be well documented and include a note about why we are excluding the data dir

Reply:
(WRT your overall comment: right now, changes to the data dir won't trigger a patch)
I think `TrussSpec` has a place primarily as an accessor for the Truss entity so I feel conflicted about moving hashing logic there. Does having a better default for the ignore patterns that is owned by the `TrussSpec` to ignore the data dir make more sense here? We can then document why we exclude and pass that pattern to the hashing functions.

Reply:
Cool -- yeah I think this is better, I think my main concern was the code duplicatoin & I think that having the defualt for the ignore patterns solves that